src/core/handshaker/http_connect/http_proxy_mapper.cc

```diff
@@ -132,30 +132,15 @@ std::optional<std::string> GetHttpProxyServer(
     LOG(ERROR) << "'" << uri->scheme() << "' scheme not supported in proxy URI";
     return std::nullopt;
   }
-  // Split on '@' to separate user credentials from host
-  char** authority_strs = nullptr;
-  size_t authority_nstrs;
-  gpr_string_split(uri->authority().c_str(), "@", &authority_strs,
-                   &authority_nstrs);
-  CHECK_NE(authority_nstrs, 0u);  // should have at least 1 string
-  std::optional<std::string> proxy_name;
-  if (authority_nstrs == 1) {
-    // User cred not present in authority
-    proxy_name = authority_strs[0];
-  } else if (authority_nstrs == 2) {
-    // User cred found
-    *user_cred = authority_strs[0];
-    proxy_name = authority_strs[1];
-    VLOG(2) << "userinfo found in proxy URI";
-  } else {
-    // Bad authority
-    proxy_name = std::nullopt;
+  if (uri->host_port().empty()) {
+    LOG(ERROR) << "'" << uri->host_port() << "' host is not present";
```

Comment:
There's not much point in including the host_port in the error, since it will always be empty here.